hpd_ws/util.py

```
@@ -291,3 +263,303 @@ def merge_dicts(d1, d2):
+
+        # Filters...
+        if 'filters' in cp:
+            fltrs = cp['filters']
+            if type(fltrs) is not list:
+                raise TypeError('Filters must be in a list')
+
+            for n, f in enumerate(fltrs):
+                if type(f) is not dict:
+                    raise TypeError('Filter #%d info not in a dict'
+                                    % n)
+                if 'class' not in f:
+                    raise KeyError('Filter #%d class missing' % n)
+
+                if 'id' not in f:
+                    raise KeyError('Filter #%d id missing' % n)
+                else:
+                    if not isinstance(f['id'], numbers.Integral):
+                        raise TypeError(
+                            'Filter #%d id must be integer: %s' % (n, f['id'])
+                        )
+                    elif f['id'] <= 0:
+                        raise ValueError(
+                            'Filter #%d id must be positive: %d' % (n, f['id'])
+                        )
+
+                fcls = f['class']
+                if fcls == 'H5Z_FILTER_DEFLATE':
+                    if 'level' not in f:
+                        raise KeyError('Filter #%d missing deflate level' % n)
+
+                    if not isinstance(f['level'], numbers.Integral):
+                        raise TypeError(
+                            'Filter #%d deflate level must be integer: %s'
+                            % (n, f['level'])
+                        )
+                    elif f['level'] <= 0:
+                        raise ValueError(
+                            'Filter #%d deflate level must be positive: %d'
+                            % (n, f['level'])
+                        )
+
+                elif fcls == 'H5Z_FILTER_FLETCHER32':
+                    pass
+
+                elif fcls == 'H5Z_FILTER_NBIT':
+                    pass
+
+                elif fcls == 'H5Z_FILTER_SCALEOFFSET':
```

This is a lossy filter so it should check if H5SZ_FILTER_FLETCHER32 is used or not. If FLETCHER32 is used, it should throw an error.